### DISCUSSION TITLE
What flannel-backend are you using when deploying k3s?

### DISCUSSION THREAD

**Discussion by manuelbuil** [k3s-io/k3s#6170] [Polls]
We would like to understand what flannel-backend is people using

**Comment by ChristianCiach** [comment]
I voted wireguard, because this is what our customers demand. But whenever I can get away with it, I use `host-gw` for its simplicity.

**Reply by carloscarnero** [reply]
Exactly my experience. Of course, it is very easy for me, as I'm in a position where all five of my clusters go by with host-gw. That has proven a (real tiny) bit more performant but, more importantly, it has shielded us from distro-specific problems with other backends.

**Reply by Barth-cyber** [reply]
Awesome

**Reply by Barth-cyber** [reply]
We seem to be speaking the same language. Simplicity combines with functionality is key. Lol

**Reply by ChristianCiach** [reply]
I like to use `host-gw` mostly because it makes it slightly easier to make sense of the routing tables (and to debug network related issues in general).

**Reply by cchangwen** [reply]
`host-gw` But why is k3s not managing the routes?
I need to manually add some to make the network work correctly.

**Reply by brandond** [reply]
@cchangwen because they're all supposed to be on the same l2 network? This document is linked from the k3s docs: 
https://github.com/flannel-io/flannel/blob/master/Documentation/backends.md#host-gw

**Reply by cchangwen** [reply]
#3412
